modules/cats/rules/src/main/scala/org/typelevel/fix/As.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.fix
+
+import scalafix.v1._
+import scala.meta._
+
+class As extends SemanticRule("As") {
+  override def fix(implicit doc: SemanticDocument): Patch =
+    doc.tree.collect {
+      // fa.map(_ => ())
+      case tree @ AnonymousMap(Lit.Unit()) => Patch.lint(VoidDiagnostic(tree))
+      // fa.map(_ => f)
+      case tree @ AnonymousMap(_) => Patch.lint(AsDiagnostic(tree))
```

Comment:
Like @armanbilge says I think we have to be careful here, and without backing up out of the current tree it might be hard to tell whether the `_` argument to our `AnonymousMap` here has already been evaluated.

It's a shame that `as` is not by-name for this reason.

What do you think about making this case only match on `Lit` subtypes?

Reply:
Great thank's for the feed back, I'll give that a go.